File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let maxArea = 0;
    let left = 0;
    let right = height.length - 1;
    
    while (left < right) {
        let waterHeight = Math.min(height[left], height[right]);
        let currArea = (right - left) * waterHeight;
        
        if (height[left] <= height[right]) {
            left++;
        } else {
            right--;
        }
        maxArea = Math.max(currArea, maxArea)
        // console.log(maxArea)
    }
    return maxArea
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_water = 0
        left = 0
        right = len(height) - 1
        
        while left < right:
            shorter = min(height[right], height[left])
            area = (right - left) * shorter
            
            if area > max_water:
                max_water = area
                
            if height[left] >= height[right]:
               right -= 1
            else:
                left += 1
                

        return max_water